pkg/kapp/resources/resources.go

```diff
@@ -293,7 +302,7 @@ func (c *ResourcesImpl) Update(resource Resource) (Resource, error) {
 	return NewResourceUnstructured(*updatedUn, resType), nil
 }
 
-func (c *ResourcesImpl) Patch(resource Resource, patchType types.PatchType, data []byte) (Resource, error) {
+func (c *ResourcesImpl) Patch(resource Resource, patchType types.PatchType, data []byte, dryRun bool) (Resource, error) {
```

Comment:
I wonder if we  should add something create a struct `resource.PatchOpts` and pass that to the function instead.
Something like 
```go
type PatchOpts struct {
    dryRun bool
}
```
This allows us to add options in the future easily without changing the function definition. 🤔

Reply:
Done